File: ONLINE_CODING_TEST/school.programmers/lvl_2/12973_짝지어_제거하기.java
import java.util.Stack;

public class Solution {
    public static int solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for (char c : s.toCharArray()) {
            if (!stack.isEmpty() && stack.peek() == c) stack.pop();
            else stack.push(c);
        }
        
        return stack.isEmpty() ? 1 : 0;
    }
    
    public static void main(String args[]) {
        System.out.println(solution("baabaa"));
        System.out.println();
        System.out.println(solution("cdcd"));
    }
}

File: ONLINE_CODING_TEST/school.programmers/lvl_2/12973_짝지어_제거하기.js
function solution(s) {
  const stack = [];

  for (let i = 0; i < s.length; i++) {
    if (!stack.length || stack[stack.length - 1] !== s[i]) stack.push(s[i]);
    else stack.pop();
  }

  return stack.length ? 0 : 1;
}

const tests = ['baabaa', 'cdcd'];

for (const test of tests) {
  console.log('-'.repeat(100));
  console.log('param : ', test, '\n');
  console.time();
  console.log('\n\t->\t', solution(test));
  console.timeEnd();
  console.timeLog();
  console.log('-'.repeat(100) + '\n');
}